src/Enforcers/ModelValidationEnforcerTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+use PDGA\DataObjects\Attributes\Column;
+use PDGA\DataObjects\Enforcers\ModelValidationEnforcer;
+use PDGA\DataObjects\Validators\EmailValidator;
+use PDGA\DataObjects\Validators\MaxLengthValidator;
+
+class NamedPerson
+{
+    #[Column(
+        name: 'email',
+        maxLength: 30
+    )]
+    #[MaxLengthValidator(15), EmailValidator]
+    public ?string $email;
+    public int $id;
+    #[Column(
+        name: 'name',
+        maxLength: 20
+    )]
+    #[MaxLengthValidator(15)]
+    public string $name;
+}
+
+class ModelValidationEnforcerTest extends TestCase
+{
+    private $enforcer;
+
+    protected function setUp(): void
+    {
+        $this->enforcer = new ModelValidationEnforcer();
+    }
+
+    public function testMetadataReturnsCorrectNumberOfValidators(): void
+    {
+        $result = $this->enforcer->getValidationMetadata(NamedPerson::class);
+
+        //id validators include int and not null
+        $this->assertEquals(count($result['id']['validators']), 2);

```

Comment:
How does `$id` have validators if it doesn't have a Column attribute? Or am I missing something?

Reply:
ModelValidationEnforcer extends ValidationEnforcer.  The ModelValidationEnforcer metadata function calls the parent function first.

Reply:
I see. All properties are validated based on their type at least, regardless of whether they have a `Column` attribute.